src/main/java/net/fabricmc/meta/utils/LoaderMeta.java

```diff
@@ -17,21 +18,25 @@
 package net.fabricmc.meta.utils;
 
 import com.google.gson.JsonObject;
+import net.fabricmc.meta.data.VersionDatabase;
 import net.fabricmc.meta.web.WebServer;
 import net.fabricmc.meta.web.models.LoaderInfoBase;
+import net.fabricmc.meta.web.models.MavenBuildVersion;
 import org.apache.commons.io.FileUtils;
 
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.FileReader;
 import java.io.IOException;
 import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class LoaderMeta {
 
 	public static final File BASE_DIR = new File("metadata");
 	public static final String MAVEN_URL = "https://maven.fabricmc.net/";
```

Comment:
what is upstream maven used for here?

Reply:
For getting available upstream loader versions.